modules/axtask/src/task.rs

```diff
@@ -266,6 +310,50 @@ impl TaskInner {
         self.in_timer_list.store(in_timer_list, Ordering::Release);
     }
 
+    /// Returns current available timer ticket ID.
+    #[inline]
+    #[cfg(feature = "irq")]
+    pub(crate) fn timer_ticket(&self) -> usize {
+        self.timer_ticket_id.load(Ordering::Acquire)
+    }
+
+    /// Expire one timer ticket ID, increment timer_ticket_id by 1,
+    /// which can be used to identify one timer event is triggered or expired.
+    #[inline]
+    #[cfg(feature = "irq")]
+    pub(crate) fn timer_ticket_expire_one(&self) {
+        self.timer_ticket_id.fetch_add(1, Ordering::Release);
+    }
+
+    pub(crate) fn unblock_locked<F>(&self, mut run_queue_push: F)
+    where
+        F: FnMut(),
+    {
+        debug!("{} unblocking", self.id_name());
+
+        // If the owning (remote) CPU is still in the middle of schedule() with
+        // this task as prev, wait until it's done referencing the task.
+        //
+        // Pairs with the `clear_prev_task_on_cpu()`.
+        //
+        // This ensures that tasks getting woken will be fully ordered against
+        // their previous state and preserve Program Order.
+        if self.on_cpu() {
```

Comment:
it refers to implementations in skyloft
```C
    /* task must be scheduled atomically */
    if (unlikely(atomic_load_acq(&next->stack_busy))) {
        /* wait until the scheduler finishes switching stacks */
        while (atomic_load_acq(&next->stack_busy)) cpu_relax();
    }
```

And I think it can be cooperated with the assertion in the end of this `if` code block